csvqb/csvqb/models/cube/qb/components/codelist.py

```diff
@@ -56,6 +62,55 @@ def __hash__(self):
         return self.code.__hash__()
 
 
+@dataclass
+class NewQbCodeListInCsvW(QbCodeList):
+    """
+    Contains the reference to an existing skos:ConceptScheme defined in a CSV-W.
+    """
+
+    schema_metadata_file_path: Path
+    csv_file_relative_path_or_uri: str = field(init=False, repr=False)
+    concept_scheme_uri: str = field(init=False, repr=False)
+    concept_template_uri: str = field(init=False, repr=False)
+
+    _schema_metadata_file_path_validator = validate_file_exists(
+        "schema_metadata_file_path"
+    )
+
+    @root_validator(pre=True)
+    def _csvw_contains_sufficient_information_validator(cls, values: dict) -> dict:
+        csv_path = values.get("csv_file_relative_path_or_uri")
+        cs_uri = values.get("concept_scheme_uri")
+        c_template_uri = values.get("concept_template_uri")
+        if csv_path is None or cs_uri is None or c_template_uri is None:
+            schema_metadata_file_path = values["schema_metadata_file_path"]
+            # The below should throw an exception if there is any problem with the CSV-W.
+            extract_code_list_concept_scheme_info(schema_metadata_file_path)
+
+            # if there's no exception but the values aren't set, something weird has happened.
+            raise ValueError(
```

Comment:
IMO this is a helpful error message. @rossbowen, could you let us know whether these kind of messages are the form you want when csvwlib is validating?

Reply:
I was thinking about this the other day, I think we could do with having two error messages in the `ValidationError` class. One which is user-friendly and another which contains more technical information.

It might be worth creating a ticket to ensure we're doing something like this before we release publically.

Reply:
Sounds like a good issue to raise. ;)